modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/Commands.scala

```diff
@@ -0,0 +1,172 @@
+package edu.gemini.seqexec.server
+
+import edu.gemini.pot.sp.SPObservationID
+import edu.gemini.spModel.`type`.{DisplayableSpType, LoggableSpType, SequenceableSpType}
+import edu.gemini.spModel.config2.{ConfigSequence, ItemKey}
+import edu.gemini.spModel.core.Peer
+
+import scalaz.Scalaz._
+import scalaz._
+
+sealed trait CommandError {
+  val msg: String
+}
+case class BadParameter(msg: String) extends CommandError
+case class ObsIdNotFound(msg: String) extends CommandError
+case class SeqexecFailureError(e: SeqexecFailure) extends CommandError {
+  val msg = SeqexecFailure.explain(e)
+}
+
+case class CommandResponse(msg: String, keys: List[(String, String)], steps: List[String])
+
+object CommandResponse {
+  def apply(msg: String): CommandResponse = CommandResponse(msg, Nil, Nil)
+}
+
+sealed trait Commands {
+  def host(): CommandError \/ CommandResponse
+  def host(s: String): CommandError \/ CommandResponse
+  def showCount(obsId: String): CommandError \/ CommandResponse
+  def showStatic(obsId: String): CommandError \/ CommandResponse
+  def showStatic(obsId: String, system: String): CommandError \/ CommandResponse
+  def showDynamic(obsId: String, step: String): CommandError \/ CommandResponse
+  def showDynamic(obsId: String, step: String, system: String): CommandError \/ CommandResponse
+  def run(obsId: String): CommandError \/ CommandResponse
+  def stop(obsId: String): CommandError \/ CommandResponse
+  def continue(obsId: String): CommandError \/ CommandResponse
+  def state(obsId: String): CommandError \/ CommandResponse
+}
```

Comment:
I think I would be tempted to give `CommandError \/ CommandResponse` a type alias like

````scala
type CommandResult[A] = CommandError \/ A
````

but that's just a personal preference.  The more explicit form could be considered clearer.